contracts/defi/provider/CompoundV2Provider.sol

```diff
@@ -0,0 +1,405 @@
+     */
+    function addDebt(
+        BaseWallet _wallet, 
+        bytes32 _loanId, 
+        address _debtToken, 
+        uint256 _debtAmount, 
+        address[] calldata _oracles
+    ) 
+        external 
+    {
+        require(_oracles.length == 2, "CompoundV2: invalid oracles length");
+        address dToken = CompoundRegistry(_oracles[1]).getCToken(_debtToken);
+        address comptroller = _oracles[0];
+        enterMarketIfNeeded(_wallet, dToken, comptroller);
+        borrow(_wallet, dToken, _debtAmount);
+    }
+
+    /**
+     * @dev Decreases the debt by repaying some token from a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the loan if any, 0 otherwise.
+     * @param _debtToken The token to repay.
+     * @param _debtAmount The amount of token to repay.
+     * @param _oracles (optional) The address of one or more oracles contracts that may be used by the provider to query information on-chain.
+     */
+    function removeDebt(
+        BaseWallet _wallet, 
+        bytes32 _loanId, 
+        address _debtToken, 
+        uint256 _debtAmount, 
+        address[] calldata _oracles
+    ) 
+        external
+    {
+        require(_oracles.length == 2, "CompoundV2: invalid oracles length");
+        address dToken = CompoundRegistry(_oracles[1]).getCToken(_debtToken);
+        repayBorrow(_wallet, dToken, _debtAmount);
+        address comptroller = _oracles[0];
+        exitMarketIfNeeded(_wallet, dToken, comptroller);
+    }
+
+    /**
+     * @dev Gets information about a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the loan if any, 0 otherwise.
+     * @param _oracles (optional) The address of one or more oracles contracts that may be used by the provider to query information on-chain.
+     * @return a status [0: no loan, 1: loan is safe, 2: loan is unsafe and can be liquidated] and the estimated ETH value of the loan
+     * combining all collaterals and all debts. When status = 1 it represents the value that could still be borrowed, while with status = 2
+     * it represents the value of collateral that should be added to avoid liquidation.      
+     */
+    function getLoan(
+        BaseWallet _wallet, 
+        bytes32 _loanId, 
+        address[] calldata _oracles
+    ) 
+        external 
+        view 
+        returns (uint8 _status, uint256 _ethValue)
+    {
+        address comptroller = _oracles[0];
+        (uint error, uint liquidity, uint shortfall) = Comptroller(comptroller).getAccountLiquidity(address(_wallet));
+        require(error == 0, "Compound: failed to get account liquidity");
+        if(liquidity > 0) {
+            return (1, liquidity);
+        }
+        if(shortfall > 0) {
+            return (2, shortfall);
+        }
+        return (0,0);
```

Comment:
I think we are missing the (rare) case where the loan exists but both `liquidity` and `shortfall` are `0`, in which case we should return `(1, 0)`.

Reply:
Yes probably, but this is a limitation from Compound. If there is no loan (i.e. no entered markets) the Comptroller will return (0,0,0) and we have no way to discriminate between 'no loan' and 'a loan but no liquidity and no shortfall'. I believe the first scenario is much more likely in our case hence we must miss the edge case you describe.